src/main/java/de/tum/cit/aet/artemis/core/service/telemetry/TelemetrySendingService.java

```diff
@@ -0,0 +1,131 @@
+     * and optionally, administrator details. If Eureka is enabled, the number of registered
+     * instances is also included.
+     *
+     * <p>
+     * The method constructs the telemetry data object, converts it to JSON, and sends it to a
+     * telemetry collection server. The request is sent asynchronously due to the {@code @Async} annotation.
+     *
+     * @param eurekaEnabled    a flag indicating whether Eureka is enabled. If {@code true},
+     *                             the method retrieves the number of instances registered with Eureka.
+     * @param sendAdminDetails a flag indicating whether to include administrator details in the
+     *                             telemetry data (such as contact information and admin name).
+     */
+    @Async
+    public void sendTelemetryByPostRequest(boolean eurekaEnabled, boolean sendAdminDetails) {
+        try {
+            String telemetryJson = new ObjectMapper().writer().withDefaultPrettyPrinter().writeValueAsString(buildTelemetryData(sendAdminDetails, eurekaEnabled));
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            HttpEntity<String> requestEntity = new HttpEntity<>(telemetryJson, headers);
+
+            log.info("Sending telemetry to {}", destination);
+            var response = restTemplate.postForEntity(destination + "/api/telemetry", requestEntity, String.class);
+            log.info("Successfully sent telemetry data. {}", response.getBody());
+        }
+        catch (JsonProcessingException e) {
+            log.warn("JsonProcessingException in sendTelemetry.", e);
+        }
+        catch (Exception e) {
+            log.warn("Exception in sendTelemetry, with dst URI: {}", destination, e);
+        }
+    }
+
+    /**
+     * Retrieves telemetry data for the current system configuration, including details
+     * about the active profiles, data source type, and optionally admin contact details.
+     *
+     * @param sendAdminDetails whether to include admin contact information in the telemetry data
+     * @param isMultiNode      whether the application runs in multi node or single node mode
+     * @return an instance of {@link TelemetryData} containing the gathered telemetry information
+     */
+    private TelemetryData buildTelemetryData(boolean sendAdminDetails, boolean isMultiNode) {
+        TelemetryData telemetryData;
+        var dataSource = datasourceUrl.startsWith("jdbc:mysql") ? "mysql" : "postgresql";
+        List<String> activeProfiles = Arrays.asList(env.getActiveProfiles());
+
+        String contact = null;
+        String adminName = null;
+        if (sendAdminDetails) {
+            contact = operatorContact;
+            adminName = operatorAdminName;
+        }
+        telemetryData = new TelemetryData(version, serverUrl, operator, activeProfiles, profileService.isProductionActive(), dataSource, isMultiNode, buildAgentCount, contact,
```

Comment:
Just as information: Our test servers also use the prod profile (to be the same as our production systems :D

Reply:
Yes, ours do that too :D I'll also add a field to know if the server is a test server.